_docs/src/documentation/rex_pipeline.md

```diff
@@ -105,40 +105,70 @@ It'll scan for all sharpmake files recursively in 3 different folders
 
 all these files are then passed on to sharpmake and executed.
 
-The generation script also has some config flags to enable or disable some behavior done in the generation step.
+Because the generation script is a wrapper around a sharpmake call, the script itself doesn't own the commandline arguments, these are owned by Sharpmake.
+We needed a way to let sharpmake communicate with the python script and back so that we don't have to maintain the same list of commandlines twice, once in sharpmake and once in the python script. We've achieved this by creating a `default_config.json` file that lives next to the sharpmake build sources
 
-| argument                  |  description 
-|--                         |--:            
-| -no_clang_tools           | disables clang tools in any post build command  
-| -unit_tests               |  also generates unit test projects 
-| -coverage                 |  adds code coverage flags to compilation steps 
-| -asan                     | adds code address sanitizer to compilation steps  
-| -ubsan                    | adds code undefined behavior sanitizer to compilation steps  
-| -fuzzy                    |  adds code fuzzy testing to compilation steps 
-| -sharpmake_args           | pass in additional arguments to sharpmake  
+This config file is parsed by the python script and it'll create commandline argument for them. the script will create a `config.json` file in the intermediate build folder which is passed to sharpmake. It'll read the config file and use it to initialize the generation pipeline.
```

Comment:
Why can't sharpmake read and parse the default config file as well?
This removes the dependency between them.

Reply:
I'm not sure what you mean? the config file is meant to be a data file about what can be configured and how it can be configured. the python pipeline uses this to set up its configuration and saves that to disk. this file then gets read by sharpmake to configure it

Reply:
Then maybe I don't understand what is going on.
There is a list with arguments and you put them into a file because you don't want to maintain them twice. (once in python and once in sharpmake) I get that.

But why can't both sharpmake and python read this file, parse it? Why does python need to parse it, export another file so sharpmake can load in a different file.